lib/util/src/workqueue.c

```diff
@@ -396,6 +483,7 @@ queue_delayed_work(struct workqueue_struct *wq, struct delayed_work *dwork, unsi
         list_add_tail(&dwork->work.entry, &wq->wq_delayed);
         dwork->timer.expires = expires;
         dwork->wq = wq;
+        ++wq->wq_refcnt;
```

Comment:
Was this a bug that we didn't increment the ref for a delayed work?

Reply:
Well, the ref counts are new with my patch so that I could defer destroying the workqueue while there was pending work (previously we just returned w/o destroying the workqueue if there was pending work).  Technically the caller should cancel all pending work before calling destroy, but now there could be a lingering delayed grow operation that the caller has no control over, so I added the ref counts to ensure they got waited on.

Reply:
Ah, got it. Thanks!

Reply:
So, previously both destroy_workqueue() and hse_timer_fini() would not wait for pending delayed work, which usually only happened if there was a bug.  With this patch they both now will hang until all the pending delated work completes, which might be never.  In a non-release build hse_timer_fini() will assert if there is pending delayed work so that we can fix the problem, but in release build it will wait (although I accelerate it so that it will expire on the next jclock tick).